protocols/gossipsub/src/handler.rs

```diff
@@ -564,17 +516,58 @@ impl ConnectionHandler for Handler {
             Self::OutboundOpenInfo,
         >,
     ) {
+        if event.is_outbound() {
+            self.outbound_substream_establishing = false;
+        }
+
+        if event.is_inbound() && self.inbound_substreams_created == MAX_SUBSTREAM_CREATION {
```

Comment:
I typically prefer to use `>=` than an `==`. 
Although it shouldn't matter in this current instance, it can be more robust against modifications in the future, for example, someone incrementing inbound_substreams_created in the `poll()`, which causes us to exceed the limit.